cmd/swarm/config.go

```diff
@@ -126,7 +127,8 @@ func buildConfig(ctx *cli.Context) (config *bzzapi.Config, err error) {
 func initSwarmNode(config *bzzapi.Config, stack *node.Node, ctx *cli.Context, nodeconfig *node.Config) error {
 	//at this point, all vars should be set in the Config
 	//get the account for the provided swarm account
-	prvkey := getAccount(config.BzzAccount, ctx, stack)
+	bzzaccount, prvkey := getAccount(config.BzzAccount, ctx, stack)
+	config.BzzAccount = bzzaccount
```

Comment:
I hate this, but we are already doing it below - modifying input arguments to this function - namely `config`. I think this is a mess and should be improved at some point.

If you just see this function, you have no ideas it has such effects.

Reply:
Yeah, I agree that this is ugly and should be refactored whenever we do some major changes to how we parse the configuration and start a swarm node based on the config.